### PR TITLE
chore: cleanup build flags

### PR DESCRIPTION
**Pull request by vitorhugods**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [X] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [X] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [X] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Build flags are a bit all over the place and it's hard to follow what's actually implemented and supported VS what was just copy/pasted from the Scala codebase.

### Solutions

- Unused build flags removed.
- Standardise casing of all remaining.
- Attempt to group into categories.
- Add simple documentation

If we need to support new build configs, then we can just add it 🤷🏻‍♂️ 

----
#### PR Post Merge Checklist for internal contributors

- [X] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1701/2/) **failed**.

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1701/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.1.0(6313437)(6313437)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.1.0(6313437)(6313437)-staging-compat.apk)

**Review by MohamadJaara — APPROVED** [review]

**Review by trOnk12 — APPROVED** [review]
